compat/cpu.h

```diff
@@ -9,23 +9,34 @@
 #ifndef SQUID_COMPAT_CPU_H
 #define SQUID_COMPAT_CPU_H
 
-#if HAVE_ERRNO_H
-#include <errno.h> /* for ENOTSUP */
-#endif
 #if HAVE_SCHED_H
 #include <sched.h>
 #endif
 
-#if !HAVE_CPU_AFFINITY
-/* failing replacements to minimize the number of if-HAVE_CPU_AFFINITYs */
-#if !HAVE_CPU_SET_T
+#if !HAVE_SCHED_SETAFFINITY
+
+#if HAVE_STDBOOL_H
+#include <stdbool.h>
+#endif
+#if HAVE_SYS_PARAM_H
+#include <sys/param.h>
+#endif
+
+#if HAVE_SYS_CPUSET_H
+#include <sys/cpuset.h>
+#if HAVE_CPUSET_T
```

Comment:
The hidden assumptions and dependencies in the above code make it very difficult to understand and maintain IMO. I rewrote this comment several times because it is difficult for me to tease out the exact problems -- they are all intermingled with those implicit assumptions. At the end, I decided to focus on one specific bug (which is not even visible in the diff) for now:

```C++
// glibc prior to 2.6 lacks CPU_COUNT
...
#define CPU_COUNT(set) CpuCount(set)
...

// glibc prior to 2.7 lacks CPU_AND
...
#define CPU_AND(destset, srcset1, srcset2) CpuAnd((destset), (srcset1), (srcset2))
...
```

AFAICT, the above (and other) definitions existed, in part, to provide missing macros in Linux environments where HAVE_SCHED_SETAFFINITY is true. Now, those definitions got secretly moved into a barely visible "!HAVE_SCHED_SETAFFINITY && !HAVE_SYS_CPUSET_H" section which has nothing to do with those Linux environments.

Again, it is difficult for me to give specific advice on how to fix this because I have problems with the overall approach. If you are tired of fixing this PR, please let me know. I will try to find the time to rewrite these sections for testing on your side.